Servercrash clientside

https://bytebin.lucko.me/OnrPZoYrvc

```
Incompatible mod set!
net.fabricmc.loader.impl.FormattedException: net.fabricmc.loader.impl.discovery.ModResolutionException: Mod discovery failed!
	at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:192)
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:142)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:71)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: net.fabricmc.loader.impl.discovery.ModResolutionException: Mod discovery failed!
	at net.fabricmc.loader.impl.discovery.ModDiscoverer.lambda$discoverMods$1(ModDiscoverer.java:137)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.discovery.ModDiscoverer.discoverMods(ModDiscoverer.java:137)
	at net.fabricmc.loader.impl.FabricLoaderImpl.setup(FabricLoaderImpl.java:208)
	at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:187)
	... 3 more
Caused by: net.fabricmc.loader.impl.metadata.ParseMetadataException: Error reading fabric.mod.json file for mod at C:\Users\elias\curseforge\minecraft\Instances\Deathdusk\mods\InvalidEnchantDetector-1.0-SNAPSHOT.jar: Invalid mod id InvalidEnchantDetector:
	- It starts with an invalid character 'I' (it must be a lowercase a-z - uppercase isn't allowed anywhere in the ID)
	- It contains invalid characters: 'D', 'E'!
	at net.fabricmc.loader.impl.metadata.ModMetadataParser.checkModId(ModMetadataParser.java:233)
	at net.fabricmc.loader.impl.metadata.ModMetadataParser.parseMetadata(ModMetadataParser.java:54)
	at net.fabricmc.loader.impl.discovery.ModDiscoverer$ModScanTask.computeJarFile(ModDiscoverer.java:269)
	at net.fabricmc.loader.impl.discovery.ModDiscoverer$ModScanTask.compute(ModDiscoverer.java:232)
	at net.fabricmc.loader.impl.discovery.ModDiscoverer$ModScanTask.compute(ModDiscoverer.java:194)
	at java.base/java.util.concurrent.RecursiveTask.exec(RecursiveTask.java:100)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

FIX YOUR MOD!! Lol.

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,7 +1,7 @@
 # Done to increase the memory available to gradle.
 org.gradle.jvmargs=-Xmx3G
 # Mod Properties
-mod_version=1.6.0
+mod_version=1.6.2
 maven_group=io.github.llamarama.team
 archives_base_name=Llamarama
 # Minecraft/Fabric Loader deps

--- a/src/main/java/io/github/llamarama/team/common/util/TradeUtil.java
+++ b/src/main/java/io/github/llamarama/team/common/util/TradeUtil.java
@@ -184,7 +184,7 @@ public final class TradeUtil {
 
             List<EnchantmentLevelEntry> finalEnchants = appliedEnchantments.stream()
                 .map(it -> new EnchantmentLevelEntry(it,
-                    random.nextInt(Math.max(0, it.getMaxLevel())) + 1))
+                    random.nextInt(Math.max(1, it.getMaxLevel())) + 1))
                 .toList();
 
             ItemStack bookOut = Items.ENCHANTED_BOOK.getDefaultStack();

--- a/src/main/java/io/github/llamarama/team/common/util/TradeUtil.java
+++ b/src/main/java/io/github/llamarama/team/common/util/TradeUtil.java
@@ -58,7 +58,12 @@ public final class TradeUtil {
         new DefaultTrade(Items.EMERALD, 2, Items.IRON_INGOT, 4),
         new DefaultTrade(Items.EMERALD, 2, Items.GOLD_INGOT, 4)
     };
-    private static final List<Enchantment> ENCHANTMENTS = Registry.ENCHANTMENT.stream().toList();
+
+    // Handling maxLevel <= 0 as a common practice for disabling enchantments.
+    // Look here: https://github.com/StrikerRockers-Mods/VanillaTweaks/issues/164
+    // And here: https://github.com/LlamaramaTeam/Llamarama/issues/15
+    private static final List<Enchantment> ENCHANTMENTS =
+        Registry.ENCHANTMENT.stream().filter(it -> it.getMaxLevel() > 0).toList();
 
     public record SellItemForEmeralds(ItemStack sell, int maxEmeralds, int mixEmeralds) implements TradeOffers.Factory {
 
@@ -183,8 +188,10 @@ public final class TradeUtil {
             }
 
             List<EnchantmentLevelEntry> finalEnchants = appliedEnchantments.stream()
-                .map(it -> new EnchantmentLevelEntry(it,
-                    random.nextInt(Math.max(1, it.getMaxLevel())) + 1))
+                .map(it -> new EnchantmentLevelEntry(
+                    it,
+                    random.nextInt(it.getMaxLevel() + (random.nextBoolean() ? 1 : 0)))
+                )
                 .toList();
 
             ItemStack bookOut = Items.ENCHANTED_BOOK.getDefaultStack();

--- a/src/main/java/io/github/llamarama/team/common/util/TradeUtil.java
+++ b/src/main/java/io/github/llamarama/team/common/util/TradeUtil.java
@@ -94,7 +94,7 @@ public final class TradeUtil {
                     ItemStack out = new ItemStack(Items.ENCHANTED_BOOK);
                     EnchantedBookItem.addEnchantment(out, new EnchantmentLevelEntry(
                         selectedEnchantment,
-                        Math.max(selectedEnchantment.getMaxLevel(), 0)
+                        selectedEnchantment.getMaxLevel()
                     ));
                     ItemStack buy = new ItemStack(Items.EMERALD, MathHelper.nextInt(random, 5, 64));
 
@@ -190,7 +190,7 @@ public final class TradeUtil {
             List<EnchantmentLevelEntry> finalEnchants = appliedEnchantments.stream()
                 .map(it -> new EnchantmentLevelEntry(
                     it,
-                    random.nextInt(it.getMaxLevel() + (random.nextBoolean() ? 1 : 0)))
+                    random.nextInt(it.getMaxLevel()) + 1)
                 )
                 .toList();
 

